touchforms/formplayer/static/formplayer/script/xforgasm.js

```diff
@@ -16,8 +16,52 @@ var ERROR_MESSAGE = "Something unexpected went wrong on that request. " +
     "If you have problems filling in the rest of your form please submit an issue. " +
     "Technical Details: ";
 
+function TaskQueue() {
+    this.queue = [];
+};
+
+/*
+ * Executes the queue in a FIFO action. If name is supplied, will execute the first
+ * task for that name.
+ */
+TaskQueue.prototype.execute = function(name) {
+    var task,
+        idx;
+    if (name) {
+        idx = _.indexOf(_.pluck(this.queue, 'name'), name);
+        task = this.queue.slice(idx, idx + 1)[0];
+        this.queue = [].concat(this.queue.slice(0, idx), this.queue.slice(idx + 1));
```

Comment:
`splice`! i knew there was a better js built in function i just couldn't seem to find it at the time of writing. thanks i'll update